052-record-reconciled-versions.md

```diff
@@ -0,0 +1,190 @@
+In the potential case where a user is using multiple `KafkaNodePool`s for a cluster using different Kafka Versions, it is my recommendation that `status.kafkaVersion` is a comma delimitered (sequentially ordered) list of all these Kafka Versions rather than the minimum or maximum kafka version.
+In discussions it was deemed unlikely that Strimzi would choose to support multiple Kafka versions at once like this, but this would be my proposal for how it is displayed to the user if it was supported.
+
+## Affected/not affected projects
+
+To the best of my knowledge only the `strimzi-kafka-operator` would be affected by this mechanism, including the code changes to the reconcilers and the doc changes.
+
+## Compatibility
+
+This mechanism is backwards compatible, in the sense that it works even if the status fields were not set on prior versions or reconciles.
+No compatibility issues, unless this feature was later removed in which case a user would no longer be able to rely on this status field.
+
+## Rejected alternatives
+
+- Using two new metadata annotations both `strimzi.io/reconciled` and `strimzi.io/reconciling` annotations, which would be a two staged update, so as to distinguish between a start of a reconcile and a completed one on the current operator version. However annotations can be overwritten by users.
+
+- Just checking the pods from the StrimziPodSets, this is the current mechanism, but is hardly ideal, given there can be quite a few pods, and the KafkaAssembly already has a step to wait for all pods to be rolled, we can add the logic for updating the status to happen after this.
+
+- Having a field `Kafka.status.versions.managedBy` that is updated at the beginning of reconcile, this has little value to the user and might just cause more confusion.
+
+- It may be useful to also include versions for the `inter-broker-protocol-version` and `log-message-format-version` in the status, such as `status.versions.kafkaInterBrokerProtocol` and `status.versions.kafkaLogMessageFormat` as these too could cause multiple reconciles and a user might want to verify that these have been picked up by the brokers. However these are not used in KRaft as `log.message.format.version` is ignored if `inter.broker.protocol.version` is 3.0 or higher. `inter.broker.protocol.version` is not deprecated formally. But it is not used in KRaft mode. So it will be essentially removed in Kafka `4.0`. So removing it from this proposal.
+
+- The `Kafka.status.kafkaVersion` field being an array of `String`s for the purposes of covering the multiple-kafka versions scenario in the [KafkaNodePool section](#-`KafkaNodePool`-considerations).
+  i.e
+  ```
+  status:
+    kafkaVersion:
+    - '3.5.0'
+    - '3.5.1'
+  ```
+  since this is unlikely to be supported, and reading a string rather than an array is easier, a single string for the field seems preferable.
```

Comment:
```suggestion
Since this is unlikely to be supported, and reading a string rather than an array is easier, a single string for the field seems preferable.
```